crabs/detection_tracking/inference_model.py

```diff
@@ -0,0 +1,394 @@
+                transforms.ToTensor(),
+            ]
+        )
+        img = transform(final_frame)
+        img = img.to(device)
+
+        img = img.unsqueeze(0)
+        prediction = self.trained_model(img)
+        pred_score = list(prediction[0]["scores"].detach().cpu().numpy())
+
+        if not self.args.sort:
+            from _inference import inference_detection
+
+            frame_out = inference_detection(
+                frame, prediction, pred_score, self.score_threshold
+            )
+
+        else:
+            from _inference import inference_tracking
+
+            frame_out = inference_tracking(
+                frame,
+                prediction,
+                pred_score,
+                self.score_threshold,
+                self.sort_crab,
+            )
+        return frame_out
+
+    def _load_video(self) -> None:
+        """
+        Load the input video and perform inference on its frames.
+        """
+        video = cv2.VideoCapture(self.vid_dir)
+
+        if not video.isOpened():
+            raise Exception("Error opening video file")
+
+        frame_width = int(video.get(cv2.CAP_PROP_FRAME_WIDTH))
+        frame_height = int(video.get(cv2.CAP_PROP_FRAME_HEIGHT))
+        cap_fps = video.get(cv2.CAP_PROP_FPS)
+
+        video_file = (
+            f"{Path(self.vid_dir).parent.stem}_" f"{Path(self.vid_dir).stem}_"
+        )
+
+        output_file = f"{video_file}_output_video.mp4"
+        output_codec = cv2.VideoWriter_fourcc(*"mp4v")
+        out = cv2.VideoWriter(
+            output_file, output_codec, cap_fps, (frame_width, frame_height)
+        )
+
+        if args.image_stack:
+            frame_count = 0
+
+            # get image size
+            width = video.get(cv2.CAP_PROP_FRAME_WIDTH)
+            height = video.get(cv2.CAP_PROP_FRAME_HEIGHT)
+
+            # initialise array for mean blurred frame
+            mean_blurred_frame = np.zeros((int(height), int(width)))
+
+            # initialise array for max_abs_blurred_frame
+            max_abs_blurred_frame = np.zeros((int(height), int(width)))
+
+            while video.isOpened() and frame_count <= 500:
+                ret, frame = video.read()
+                if not ret:
+                    print("No frame read. Exiting...")
+                    break
+
+                else:
+                    # Apply transformations to the frame
+                    gray_frame, blurred_frame = self.apply_grayscale_and_blur(
+                        frame, [5, 5], 0
+                    )
+
+                    # accumulate blurred frames
+                    mean_blurred_frame += blurred_frame
+
+                    # accumulate max absolute values
+                    max_abs_blurred_frame = np.maximum(
+                        max_abs_blurred_frame, abs(blurred_frame)
+                    )
+
+                    frame_count += 1
+            # compute the mean
+            mean_blurred_frame = mean_blurred_frame / frame_count
```

Comment:
can we reuse the utils for computing the additional channels here? To avoid code duplication and making sure we are using the same approach in training and inference.

If it's a lot for this PR feel free to open an issue